Source/NSGridView.m

```diff
@@ -0,0 +1,1280 @@
+                                at: &_yPlacement];
+      [coder encodeValueOfObjCType:@encode(NSInteger)
+                                at: &_rowAlignment];
+    }
+}
+
+- (instancetype) initWithCoder: (NSCoder *)coder
+{
+  self = [super init];
+  if (self != nil)
+    {
+      NSDebugLog(@"%@ %@",NSStringFromClass([self class]), NSStringFromSelector(_cmd));
+      
+      if ([coder allowsKeyedCoding])
+        {
+          if ([coder containsValueForKey: @"NSGrid_content"])
+            {
+              [self setContentView: [coder decodeObjectForKey: @"NSGrid_content"]];
+              NSDebugLog(@"contentView = %@", [self contentView]);
+            }
+          if ([coder containsValueForKey: @"NSGrid_mergeHead"])
+            {
+              _mergeHead = [coder decodeObjectForKey: @"NSGrid_mergeHead"];
+            }
+          if ([coder containsValueForKey: @"NSGrid_owningRow"])
+            {
+              _owningRow = [coder decodeObjectForKey: @"NSGrid_owningRow"]; // weak
+              NSDebugLog(@"_owningRow = %@", _owningRow);
+            }
+          if ([coder containsValueForKey: @"NSGrid_owningColumn"])
+            {
+              _owningColumn = [coder decodeObjectForKey: @"NSGrid_owningColumn"]; // weak
+              NSDebugLog(@"_owningColumn = %@", _owningColumn);
+            }
+          if ([coder containsValueForKey: @"NSGrid_xPlacement"])
+            {
+              _xPlacement = [coder decodeIntegerForKey: @"NSGrid_xPlacement"];
+            }      
+          if ([coder containsValueForKey: @"NSGrid_yPlacement"])
+            {
+              _yPlacement = [coder decodeIntegerForKey: @"NSGrid_yPlacement"];
+            }  
+          if ([coder containsValueForKey: @"NSGrid_alignment"])
+            {
+              _rowAlignment = [coder decodeIntegerForKey: @"NSGrid_alignment"];
+            }      
+        }
+      else
+        {
+          [self setContentView: [coder decodeObject]];
+          ASSIGN(_mergeHead, [coder decodeObject]);
+          _owningRow = [coder decodeObject]; // weak
+          _owningColumn = [coder decodeObject]; // weak
+          [coder decodeValueOfObjCType: @encode(NSInteger)
+                                    at: &_xPlacement];
+          [coder decodeValueOfObjCType:@encode(NSInteger)
+                                    at: &_yPlacement];
+          [coder decodeValueOfObjCType:@encode(NSInteger)
+                                    at: &_rowAlignment];
+        }
+    }
+  return self;
+}
+
+- (void) dealloc
+{
+  RELEASE(_contentView);
+  RELEASE(_mergeHead);
+  [super dealloc];
+}
+
+@end
+
+/// Column ///
+@implementation NSGridColumn
+
++ (void) initialize
+{
+  if (self == [NSGridColumn class])
+    {
+      [self setVersion: 1];
+    }
+}
+
+- (void) setGridView: (NSGridView *)gv
+{
+  _gridView = gv;
+}
+
+- (NSGridView *) gridView
+{
+  return _gridView;
+}
+
+- (NSInteger) numberOfCells
+{
+  NSUInteger ci = [_gridView indexOfColumn: self];
```

Comment:
Don't all columns have the same amount of cells and that value is numberOfRows of the grid view?

Reply:
Since merging is possible some might have a different count.  That is why I did this this way.  I need to test on Cocoa/macOS to be sure.  Once I do I will make the appropriate changes here.

Reply:
Please test on Cocoa. I think your argument is wrong and if it where true most methods in the current implementation of NSGridView would fail. To be consistent you should rely on the equal size here as well.

Reply:
Tested on macOS... changed implementation.